src/controllers/idp.ts

```diff
@@ -45,10 +48,22 @@ export async function resolveSelfCareIdentity(
   try {
     // ensure an apim account is created for every session,
     // so that every user operation can rely on that
-    await createApimUserIfNotExists(
-      apimClient,
-      apimUserForSelfCareOrganization(selfcareIdentity.organization)
-    );
+    const maybeApimUser = lockSelfcareCreateNewApimUser
+      ? await getApimUser(
+          apimClient,
+          formatApimAccountEmailForSelfcareOrganization(
+            selfcareIdentity.organization
+          )
+        )
+      : await createApimUserIfNotExists(
+          apimClient,
+          apimUserForSelfCareOrganization(selfcareIdentity.organization)
+        );
+
+    // feature flag to lock new Selfcare users access
+    if (lockSelfcareCreateNewApimUser && isNone(maybeApimUser)) {
+      return ResponsePermanentRedirect(options.failureLoginPage);
+    }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** Do you still need to check `lockSelfcareCreateNewApimUser`? Isn't `isNone(maybeApimUser)`‏ enough to determine an error? And what if `isNone(maybeApimUser)==true`  and `lockSelfcareCreateNewApimUser==false`?

Reply:
Consider that initially there was no control: therefore anyone _(even if `none`)_ could access.
I added this double check because I'm sure that, with the feature flag active, if the user is none then the login must fail.
Otherwise _(the other combinations)_ are exactly as before and not knowing the previous reasoning I went "as safe as possible".
Do you agree?

Reply:
I understand. I actually wrote that piece of software and I didn't spot the case at the time - shame on me 👎 

Besides what has been done in the past, as you are taking ownership of the project, what do you think the application should do?

Reply:
I would actually be of the opinion that if the apim user is 'none' then the login must fail. And therefore it would be enough to strengthen the logic _(past and current)_ by adding only the check on 'isNone'.

Reply:
d99f1e9